### PR TITLE
[ui] Templar improvements and bug fixes

### LINKED ISSUE #398: [:all] Should support {{ }} in the command
This way while looking at an image,

You could...

:run site.yaml --eei {{ repository }}

Or use any other info in the content

### LINKED ISSUE #400: [ui] Does not notify when a matching action is not found
Eg :foo

### LINKED ISSUE #403: [ui] Function utils/templar uses unconfigured logger
Which produces stdout and trashes the screen

### PR DESCRIPTION
**Pull request by cidrblock**
Fixes #398 
Fixes #400
Fixes #403 

- this moves the templating of a `:` command to the UI, (pre action)
- this allows for `:run site.yml --eei {{ details.repotags[0] }}`  (running a playbook using a specific image while looking at image details (as an example)
- remove templating logic from `:open` as it is being done in the UI now
- `:template` now uses the App base class for convenience
- Notification pop-up for action not found
- Notification pop-up for templating error
- Fix templar to use the configured logger
- added integration tests focued around use of `{{ }}` in the `:` command input
- adds a ui.clear() to both `:select` and `:back` to remove some bleed from the previous screens

### CODE REVIEW

**Comment by cidrblock** [comment]
Recheck

**Comment by cidrblock** [comment]
recheck

File: ansible_navigator/actions/open_file.py [hunk 1]
```diff
@@ -48,66 +43,21 @@ def __init__(self, args):
...
-                else:
-                    self._logger.debug("something just a plain string")
-                    obj = something
-        else:
-            self._logger.debug("something not provided")
-            obj = content.showing
-        return filename, line_number, obj
-
     @staticmethod
     def _remove_dbl_un(string):
```
Comment (ganeshrn) [inline ansible_navigator/actions/open_file.py:47]: The method seems to be repeated at 2 places, probably can be move in utils file

File: tests/fixtures/integration/actions/templar/test_direct_interactive_ee.py/test/9.json [hunk 2]
```diff
@@ -0,0 +1,608 @@
...
+        "~",
+        "~",
+        "~",
+        "~",
+        "~",
+        "~",
+        "~",
+        "~",
+        "~",
+        "\"~/github/ansible-navigator/tests/fixtures/integration/actions/run/site.yaml\" 27L, 548B"
```
Comment (ganeshrn) [inline tests/fixtures/integration/actions/templar/test_direct_interactive_ee.py/test/9.json:606]: Is this line expected?

File: tests/fixtures/integration/actions/templar/test_direct_interactive_noee.py/test/5.json [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    "comment": "doc for task",
+    "additional_information": {
+        "look_fors": [
+            "module: debug"
+        ],
+        "look_nots": [],
+        "compared_fixture": false
+    },
+    "output": [
+        "DEBUG (MODULE)",
```
Comment (ganeshrn) [inline tests/fixtures/integration/actions/templar/test_direct_interactive_noee.py/test/5.json:13]: Can we use `look_fors` for check if module doc is displayed. Having a entire module doc in fixture can result in failure if the actual debug module doc changes in future ansible version?

File: ansible_navigator/actions/open_file.py (hunk 1 above)
Comment (cidrblock) [reply ansible_navigator/actions/open_file.py:47]: good idea, will do

File: tests/fixtures/integration/actions/templar/test_direct_interactive_noee.py/test/9.json [hunk 4]
```diff
@@ -0,0 +1,608 @@
...
+        "~",
+        "~",
+        "~",
+        "~",
+        "~",
+        "~",
+        "~",
+        "~",
+        "~",
+        "\"~/github/ansible-navigator/tests/fixtures/integration/actions/run/site.yaml\" 27L, 548B"
```
Comment (ganeshrn) [inline tests/fixtures/integration/actions/templar/test_direct_interactive_noee.py/test/9.json:606]: Same as above

File: tests/fixtures/integration/actions/templar/test_direct_interactive_ee.py/test/9.json (hunk 2 above)
Comment (cidrblock) [reply tests/fixtures/integration/actions/templar/test_direct_interactive_ee.py/test/9.json:606]: it's the last line of the vi screen, but we don't compare against the fixture here

File: tests/fixtures/integration/actions/templar/test_welcome_interactive_ee.py/test/10.json [hunk 5]
```diff
@@ -0,0 +1,608 @@
...
+        "~",
+        "~",
+        "~",
+        "~",
+        "~",
+        "~",
+        "~",
+        "~",
+        "~",
+        "\"~/github/ansible-navigator/tests/fixtures/integration/actions/run/site.yaml\" 27L, 548B"
```
Comment (ganeshrn) [inline tests/fixtures/integration/actions/templar/test_welcome_interactive_ee.py/test/10.json:606]: same as above

File: tests/fixtures/integration/actions/templar/test_welcome_interactive_noee.py/test/10.json [hunk 6]
```diff
@@ -0,0 +1,608 @@
...
+        "~",
+        "~",
+        "~",
+        "~",
+        "~",
+        "~",
+        "~",
+        "~",
+        "~",
+        "\"~/github/ansible-navigator/tests/fixtures/integration/actions/run/site.yaml\" 27L, 548B"
```
Comment (ganeshrn) [inline tests/fixtures/integration/actions/templar/test_welcome_interactive_noee.py/test/10.json:606]: same as above

**Review by ganeshrn — APPROVED** [review]
Added minor comments can be fixed later on. Overall +1

**Pull request merged**